app/models/user.rb

```diff
@@ -0,0 +1,17 @@
+class User < ApplicationRecord
+  VALID_EMAIL_REGEX = Settings.email_regex
+  before_save :downcase_email
+  validates :name, presence: true, length: {maximum: Settings.user.name_length}
+  validates :email, presence: true,
+    length: {maximum: Settings.user.email_length},
+    format: {with: VALID_EMAIL_REGEX},
+    uniqueness: true
+  validates :password, presence: true,
+    length: {minimum: Settings.user.password_length}
+  has_secure_password
+
+  private
+  def downcase_email
+    email.downcase!
```

Comment:
downcase và downcase khác gì nhau k ô ?

Reply:
downcase trả về chuỗi kí tự in thường mà không làm thay đổi chuỗi ban đầu
VD:
string = 'Ruby'
puts string.downcase # hiển thị: ruby
puts string # hiển thị: Ruby

downcase! trả về chuỗi kí tự in thường nhưng thay đổi chuỗi ban đầu
VD:
string = 'Ruby'
puts string.downcase! # hiển thị: ruby
puts string # hiển thị: ruby